src/pybind/mgr/mds_autoscaler/module.py

```diff
@@ -0,0 +1,157 @@
+"""
+Automatically scale MDSs based on status of the file-system using the FSMap
+"""
+
+import logging
+from typing import Optional, List
+from mgr_module import MgrModule
+from ceph.deployment.service_spec import ServiceSpec
+import orchestrator
+import copy
+
+log = logging.getLogger(__name__)
+
+
+class MDSAutoscaler(orchestrator.OrchestratorClientMixin, MgrModule):
+    """
+    MDS autoscaler.
+    """
+    def __init__(self, *args, **kwargs):
+        super(MDSAutoscaler, self).__init__(*args, **kwargs)
+        self.set_mgr(self)
+
+    def get_service(self, fs_name: str) -> Optional[List[orchestrator.ServiceDescription]]:
+        result = None
+        try:
+            service = "mds.{}".format(fs_name)
+            completion = self.describe_service(service_type='mds',
+                                               service_name=service,
+                                               refresh=True)
+            self._orchestrator_wait([completion])
+            orchestrator.raise_if_exception(completion)
+            result = completion.result
+        except Exception as e:
+            self.log.exception("{}: exception while fetching service info"
+                               .format(e))
+            pass
+        return result
+
+    def get_daemons(self, fs_name: str) -> Optional[List[orchestrator.DaemonDescription]]:
+        result = None
+        try:
+            service = "mds.{}".format(fs_name)
+            completion = self.list_daemons(service_name=service)
+            self._orchestrator_wait([completion])
+            orchestrator.raise_if_exception(completion)
+            result = completion.result
+        except Exception as e:
```

Comment:
Can this be `OrchestratorError`? Let's move this to `verify_and_manage_mds_instance`. If we can't talk to the orchestrator, the module should log the exception and do nothing further.

Reply:
fixed